File: test/TestParseStatements.java
import generated.AndroidTotemSqlParser;

import java.io.ByteArrayInputStream;

public class TestParseStatements {
    public static void main(String[] args) {
        String[] cases = {
                "SELECT\n  id,\n  name,\n  price\nFROM product;",
                "SELECT\n  name AS name,\n  (price / 7) AS usprice\nFROM product\nWHERE\n  price > 5000;",
                "SELECT\n  id AS id,\n  (price * 15) AS jpprice\nFROM product\nWHERE\n  price <= 7000;",
                "SELECT\n  jpproduct -> product -> usproduct.sales,\n  jpproduct -> product -> usproduct.name\nFROM jpproduct\nWHERE\n  id = 2;"
        };
        for (String ts : cases) {
            ByteArrayInputStream in = new ByteArrayInputStream(ts.getBytes());
            try {
                AndroidTotemSqlParser.parse(in);
            } catch (Exception e) {
                e.printStackTrace();
            }
        }
    }
}

File: test/TestParseStatements.kt
import generated.AndroidTotemSqlParser
import java.io.BufferedInputStream
import java.io.FileInputStream
import kotlin.system.exitProcess

fun testParse(name: String) {
    val fileName = "./sql/$name.sql"
    val inputStream = BufferedInputStream(FileInputStream(fileName))
    try {
        val statements = AndroidTotemSqlParser.parse(inputStream)!!
        for (statement in statements) {
            println(statement)
        }
    } catch (e: Exception) {
        e.printStackTrace()
        exitProcess(1)
    }
}

fun main() {
//    testParse("select")
//    testParse("create_class")
//    testParse("create_deputy")
//    testParse("insert")
//    testParse("update")
//    testParse("delete")
//    testParse("drop")
    testParse("all")
}